changelog.d/12738.type

```diff
@@ -0,0 +1,7 @@
+Currently when a user attempts login before registration, we respond with 403 and an empty string message.
+In cases where a user was registered and bound to a 3pid already, we return "Invalid Password" as a message.
+2 issues arise from this behavior:
+- The major one is user existence leak. Having 2 different error messages for unregistered users and registered users who've entered the wrong password essentially leaks the information about that particular user's existence.
+- While another issue is having a non-informative error message in the current approach. As the user wasn't yet registered, it has no password, so essentially no password is valid, implying the need to return "Invalid Password" instead of just "".
+
+Contributed by Daniel Aloni.
```

Comment:
```suggestion
Change error message when attempting to login to a non-existent account to match that returned when the account exists but the password is invalid. Contributed by Daniel Aloni.
```

We try and keep the changelogs to one line, but we link back to the PR so this (wonderful) write up won't be lost.

Also, can you move this file to `changelog.d/12738.misc` please? The `.type` suffix is meant to be replaced by one the ones listed in the [contributing guide](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog).